### PR TITLE
区分问题和答案文件夹, 用户指定问题时, 文件放到问题文件夹; 增加了solve #{id}的功能

### PR DESCRIPTION
**Pull request by songyzh**
方便用户做题/看答案
增加solve #{id}功能, 方便把problem中的文件转移到solution中

### CODE REVIEW

**Comment by songyzh** [comment]
增加solve #{id}功能, 方便把problem中的文件转移到solution中

File: src/main.rs [hunk 1]
```diff
@@ -18,34 +20,48 @@ fn main() {
     loop {
         println!("Please enter a frontend problem id, or \"random\" to generate a random one.");
```
Comment (aylei) [inline src/main.rs]: Shall we add a message to explain the `solve $i` command?

**Review by aylei — APPROVED** [review]
The rest LGTM

File: src/main.rs [hunk 2]
```diff
@@ -18,34 +20,48 @@ fn main() {
...
+                .parse()
+                .unwrap();
+        } else {
+            id = id_arg
+                .parse::<u32>()
+                .unwrap_or_else(|_| panic!("not a number: {}", id_arg));
+            if solved_ids.contains(&id) {
+                println!(
+                    "The problem you chose is invalid (the problem may have been initialized \
+                     or may have no rust version)."
```
Comment (aylei) [inline src/main.rs]: the error message seems not match the condition that the problem has been solved.

File: src/main.rs (hunk 1 above)
Comment (songyzh) [reply src/main.rs]: fixed in https://github.com/aylei/leetcode-rust/pull/18/commits/0d72e628b6c6b11cc5edca9725e3d505fcc7c355

File: src/main.rs (hunk 2 above)
Comment (songyzh) [reply src/main.rs]: fixed in https://github.com/aylei/leetcode-rust/pull/18/commits/125c64c39b035d4cab2687d9bc2977749c8b0a59 https://github.com/aylei/leetcode-rust/pull/18/commits/76edb8c5fb172de61a1e308d279e9b877ad32c5f https://github.com/aylei/leetcode-rust/pull/18/commits/695f53a684264b19f0d2e94541fdb18cc1f37628

**Comment by songyzh** [comment]
@aylei fixed, review plz, thx!

**Review by aylei — APPROVED** [review]
LGTM, thanks!

**Pull request merged**